service/geopmdpy/session.py

```diff
@@ -31,91 +31,563 @@
+                            units of seconds.
+
+        """
+
+        self._period = period
+        # Add one to ensure:
+        #     total_time == num_loop * period
+        # because we do not delay the start iteration
+        self._num_loop = num_period + 1
+
+    def __iter__(self):
+        """Set up a timed loop
+
+        Iteration method for timed loop.  This iterator can be used in
+        a for statement to execute the loop periodically.
+
+
+        """
+        self._loop_idx = 0
+        self._target_time = time.time()
+        return self
+
+    def __next__(self):
+        """Sleep until next targeted time for loop and update counter
+
+        """
+        result = self._loop_idx
+        if self._loop_idx == self._num_loop:
+            raise StopIteration
+        if self._loop_idx != 0:
+            sleep_time = self._target_time - time.time()
+            if sleep_time > 0:
+                time.sleep(sleep_time)
+        self._target_time += self._period
+        self._loop_idx += 1
+        return result
+
+
+class Requests:
+    """Object derived from user input that provides request information
+
+    The geopmsession command line tool parses requests for reading or
+    writing from standard input.  The Requests object holds the logic
+    for parsing the input stream upon construction.  The resulting
+    object may be iterated upon to retrieve the requested
+    signals/controls that the user would like to read/write.  The
+    Request object also provides the enum used to format signal values
+    into strings (for read mode sessions).
+
+    """
+    def __init__(self):
+        raise NotImplementedError('Requests class is an abstract base class for ReadRequests and WriteRequests')
+
+    def __iter__(self):
+        """Iterate over list of requests
+
+        """
+        yield from self._requests
+
+    def get_names(self):
+        """Get the signal or control names from each request
+
+        Returns:
+            list(string): The name of the signal or control associated
+                           with each user request.
+
+        """
+        return [rr[0] for rr in self._requests]
+
+    def iterate_stream(self, request_stream):
+        """Iterated over a stream of requests
+
+        This is a generator function that will filter out comment
+        lines and trailing white space from the input stream.  It can
+        be used to iterate over a request stream from the user for
+        either a read or write mode session.
+
+        Args:
+            request_stream (file): Stream containing requests from
+                                   user.
+
+        Returns:
+            generator: Iterate over filtered lines of the
+                       request_stream
+
+        """
+        for line in request_stream:
+            line = line.strip()
+            if line == '':
+                break
+            if not line.startswith('#'):
+                yield line
+
+
+class ReadRequests(Requests):
```

Comment:
We're doing plurals now?

Reply:
The main purpose is to iterate over the object, so plural seemed appropriate.  We could call it a ReadRequestList or ReadRequestQueue?

Reply:
Fine w/either.  I just want to be consistent with all our code and I know we've been back and forth on having plural names in the codebase in the past.